src/main/java/pl/simpleascoding/tutoringplatform/repository/AdvertisementRepository.java

```diff
@@ -1,7 +1,10 @@
 package pl.simpleascoding.tutoringplatform.repository;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import pl.simpleascoding.tutoringplatform.domain.advertisement.Advertisement;
 
 public interface AdvertisementRepository extends JpaRepository<Advertisement, Long> {
+    Page<Advertisement> findAllByAuthor_NameOrderByCreationDateDesc(String username, Pageable pageable);
```

Comment:
You can add `@PageableDefault` or `@SortDefault` annotation to your controller method instead of a long JPA method name, but it's a valid approach and you don't need to change it if you don't want to.
[Info about these annotations](https://medium.com/@rasheedamir/java-spring-default-page-default-sort-cab2f410f8b0)